src/vehicle_gui.cpp

```diff
@@ -2596,7 +2596,7 @@ struct VehicleDetailsWindow : Window {
 						(v->type == VEH_ROAD && _settings_game.vehicle.roadveh_acceleration_model != AM_ORIGINAL)) {
 					const GroundVehicleCache *gcache = v->GetGroundVehicleCache();
 					if (v->type == VEH_TRAIN && (_settings_game.vehicle.train_acceleration_model == AM_ORIGINAL ||
-							GetRailTypeInfo(Train::From(v)->railtype)->acceleration_type == 2)) {
+							Train::From(v)->GetAccelerationType() == 2)) {
```

Comment:
I've seen this magic number used a lot here to check for maglev (and your changes to use the getter function instead of checking the property directly). Maybe worth a constant while you're here?